### PR TITLE
`General`: Reverse semester sorting in course creation dialog

### PR DESCRIPTION
**Pull request by milljoniaer**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context
This tiny PR reverses the sorting of  the semester dropdown, so that recent semesters are on top. This makes it faster to find the needed one.

### Steps for Testing

Prerequisites:
- 1 Instructor

1. Create course and verify that the semesters in the semester dropdown are sorted descending.

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots
image

## Summary by CodeRabbit

- **Bug Fixes**
	- Adjusted the order in which semesters are displayed, now showing the most recent semester first.

### CODE REVIEW

**Review by nata-81 — DISMISSED** [review]
Tested on ts2 works as expected.

**Review by konrad2002 — DISMISSED** [review]
manually tested on ts2, works as expected

**Review by Strohgelaender — CHANGES_REQUESTED** [review]
client tests are failing

**Review by nityanandaz — DISMISSED** [review]
Code review: looks good

**Review by coolchock — DISMISSED** [review]
code lgtm

**Review by AntonGeTUM — DISMISSED** [review]
Tested on ts11, works as expected

File: src/main/webapp/app/course/manage/course-update.component.ts [hunk 1]
```diff
@@ -498,7 +498,7 @@ export class CourseUpdateComponent implements OnInit {
             semesters[2 * i + 1] = 'SS' + (18 + i);
             semesters[2 * i + 2] = 'WS' + (18 + i) + '/' + (19 + i);
         }
-        return semesters;
+        return semesters.reverse();
```
Comment (nityanandaz) [inline src/main/webapp/app/course/manage/course-update.component.ts:501]: Or you could build the array in reverse already :)

**Review by nityanandaz — APPROVED** [review]
Re-approve code review

**Review by nityanandaz — APPROVED** [review]
Re-approve code review

**Review by egekurt123 — APPROVED** [review]
Manually tested on ts6, works as expected

**Review by farisd16 — APPROVED** [review]
tested on ts6, semesters are sorted in descending order as expected

**Review by georgyia — APPROVED** [review]
tested on ts1, works as expected

**Review by basak-akan — APPROVED** [review]
code lgtm

**Review by MaximilianJG — APPROVED** [review]
Tested on ts2. works as expected

**Comment by milljoniaer** [comment]
Since @Strohgelaender criticised the failing client tests, which run through now, I mark this PR as ready-to-merge

**Review by MaximilianAnzinger — APPROVED** [review]

**Pull request merged**